include/irr/asset/CMTLPipelineMetadata.h

```diff
@@ -0,0 +1,107 @@
+#ifndef __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+#define __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+
+#include "irr/asset/IPipelineMetadata.h"
+
+namespace irr {
+namespace asset
+{
+
+class CMTLPipelineMetadata final : public IPipelineMetadata
+{
+public:
+    struct SMtl
+    {
+        std::string name;
+
+#include "irr/irrpack.h"
+        struct
+        {
+            //Ka
+            core::vector3df_SIMD ambient = core::vector3df_SIMD(1.f);
+            //Kd
+            core::vector3df_SIMD diffuse = core::vector3df_SIMD(1.f);
+            //Ks
+            core::vector3df_SIMD specular = core::vector3df_SIMD(1.f);
+            //Ke
+            core::vector3df_SIMD emissive = core::vector3df_SIMD(1.f);
+            //Tf
+            core::vector3df_SIMD transmissionFilter = core::vector3df_SIMD(1.f);
+            //Ns, specular exponent in phong model
+            float shininess = 32.f;
+            //d
+            float opacity = 1.f;
+            //-bm
+            float bumpFactor = 1.f;
+
+            //PBR
+            //Ni, index of refraction
+            float IoR = 1.6f;
+            //Pr
+            float roughness = 0.f;
+            //Pm
+            float metallic = 0.f;
+            //Ps
+            float sheen = 0.f;
+            //Pc
+            float clearcoatThickness = 0.f;
+            //Pcr
+            float clearcoatRoughness = 0.f;
+            //aniso
+            float anisotropy = 0.f;
+            //anisor
+            float anisoRotation = 0.f;
+            //illum - bits [0;3]
+            //map presence: bits [4;16], order in accordance with E_MAP_TYPE
+            uint32_t extra = 0u;
+        } PACK_STRUCT std140PackedData;
```

Comment:
bad packing, whole struct no alligned to 16 bytes.

Also I'd consider wrapping non-PBR (shininess, opacity, bumpfactor) and PBR (IoR, roughness, metallic, ...) in separate structs and using a `union` to not waste space (we aim to use the same data in shaders).

Reply:
i dont understand. How struct can be aligned to anything? it's just struct, not memory address

Reply:
`alignas` is possible on structs and classes in C++11, it instructs the compiler that the `this` pointer will be aligned, but most importantly it will guarantee alignment on **stack** allocated objects, and within other structs... etc.

Its specified on vectorSIMDf and matrices (by composition) ;)

Reply:
ok but why this c++ struct should have a particularly aligned address?

Reply:
because we need to assume that all the structs we "share" and make builtin; will be used to construct bigger things by extensions, other loaders and users.

Making this (and any other) struct aligned and of correct size (mult of 16), will prevent it from blowing up in our face when we decide to put it in different places, arrays, UBOs, SSBOs, PCs, etc. and the C++ won't match it byte for byte.